### PR TITLE
feat: support loading config with tsx

### PR DESCRIPTION
**Pull request by privatenumber**
#### `Commit Message Summary (CHANGELOG)`

```
leverage tsx to load TypeScript & ESM config files
```

### `Type`

- [x] Perf
- [x] Feature

### `SemVer`

- [x] Feature (:label: Minor)

### `Issues`

Follow up to https://github.com/postcss/postcss-load-config/pull/249

### `Checklist`

- [x] Lint and unit tests pass with my changes
- [x] I have added tests that prove my fix is effective/works
- [x] I have added necessary documentation (if appropriate)

cc @brc-dd

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "compilerOptions": {
-    "module": "CommonJS",
-    "esModuleInterop": true
+    "module": "Preserve",
+    "esModuleInterop": true,
+    "allowJs": true,
+    "noEmit": true
```
Comment (privatenumber) [inline tsconfig.json:6]: The previous settings were not type checking `.js` files

**Comment by brc-dd** [comment]
Yeah. tsx added support for this today 😅

**Comment by privatenumber** [comment]
Im also currently implementing a custom 'import()' with TypeScript support, so TLA would be supported too. But this would be limited to newer versions of Node that supports 'module.register()'.

**Comment by ai** [comment]
Thanks to everyone. Released in 5.1.0.

**Pull request merged**